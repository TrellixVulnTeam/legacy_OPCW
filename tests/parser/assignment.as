# Slot assignment.
# Usage-based type inference will make 'x' and 'int'.
# Because 'x' is immutable it can only be assigned to once.
let x;
x = 423;

# Mutable slot assignment.
let mut y: int;
y = 432;
y = 43;
y = 8293;
