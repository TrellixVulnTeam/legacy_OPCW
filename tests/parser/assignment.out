ModuleDecl <?>
  LocalSlotDecl <?> x
  AssignExpr <?>
    Ident <?> x
    IntegerExpr <?> 'int' 423 (10)
  LocalSlotDecl <?> mut y 'int'
  AssignExpr <?>
    Ident <?> y
    IntegerExpr <?> 'int' 432 (10)
  AssignExpr <?>
    Ident <?> y
    IntegerExpr <?> 'int' 43 (10)
  AssignExpr <?>
    Ident <?> y
    IntegerExpr <?> 'int' 8293 (10)
